### PR TITLE
Split official release workflow for binaries and docker

### PR DESCRIPTION
**Pull request by agunapal**
## Description

This PR splits the official release workflow into 2 parts
  - The original workflow will only publish binaries
  - Added a second workflow for publishing docker images
    - This is using the CI_GPU runner as self hosted runners have more disk space
    -  This will also ensure that, we can publish docker images once we confirm that PyPI indices are updated with the latest version of TorchServe

Also changed the `docker push` commands to use `os.system` calls as we want this to execute first and be blocking.

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

[Test run of Docker workflow](https://github.com/pytorch/serve/actions/runs/6486104508/job/17613580328)

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: kubernetes/kserve/docker_nightly.py
Comment (msaroufim) [inline kubernetes/kserve/docker_nightly.py:61]: os.system won't error handle so if this line fails it will just keep going

**Review by msaroufim — CHANGES_REQUESTED** [review]

File: kubernetes/kserve/docker_nightly.py
Comment (agunapal) [reply kubernetes/kserve/docker_nightly.py:61]: Reading more about subprocess, it will execute in parallel to the main process.
Also, if you check the logs here , you will see that it executed the commands after delete was done https://github.com/pytorch/serve/actions/runs/6486104508/job/17613580328

So, not sure whats the right way to handle this.

File: .github/workflows/official_release_docker.yml
Comment (agunapal) [reply .github/workflows/official_release_docker.yml:31]: I didn't follow. What are we gating on? The process would be for us to trigger the binary workflow as before, which runs pending approval. Once its complete, do manual tests. Once thats complete, trigger the docker workflow, which again runs after approval.

File: kubernetes/kserve/docker_nightly.py
Comment (msaroufim) [reply kubernetes/kserve/docker_nightly.py:61]: subprocess.call and subprocess.run should be synchronous only subprocess.Popen is async and you make it sync by calling an await

Comment (agunapal) [reply kubernetes/kserve/docker_nightly.py:61]: Unless, we actually push, I dont have a way to prove that this doesn't work.

But this is what i see in the logs.

Screenshot 2023-10-11 at 11 36 47 AM

Comment (agunapal) [reply kubernetes/kserve/docker_nightly.py:61]: I can revert the nightly to use try_and_handle as before and see what happens. But for the official release, we can use the system command. Is that fine?

File: .github/workflows/official_release_docker.yml
Comment (msaroufim) [reply .github/workflows/official_release_docker.yml:31]: I'm suggesting not to upload kserve images by default

File: kubernetes/kserve/docker_nightly.py
Comment (msaroufim) [reply kubernetes/kserve/docker_nightly.py:61]: is this the dry run?

File: .github/workflows/official_release_docker.yml
Comment (agunapal) [reply .github/workflows/official_release_docker.yml:31]: Oh.I get it. We would need another input to skip uploading docker images when we want to actually push kserve images. I will add it, but not sure if I will be able to test it before the release.

File: kubernetes/kserve/docker_nightly.py
Comment (agunapal) [reply kubernetes/kserve/docker_nightly.py]: Yes

Comment (agunapal) [reply kubernetes/kserve/docker_nightly.py]: Reverted the nightly as before. We can see if it works tonight. So far, images were not getting deleted because the prune command didnt have the `-f` option

**Review by msaroufim — APPROVED** [review]
lgtm

**Comment by agunapal** [comment]
@msaroufim `subprocess.run` worked fine in last night's run. It was blocking. So, I guess only the print statements are probably not executed in sequence. Will update the official workflow to do the same

**Pull request merged**
